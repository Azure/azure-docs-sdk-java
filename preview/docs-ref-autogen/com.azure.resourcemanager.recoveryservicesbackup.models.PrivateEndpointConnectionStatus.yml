### YamlMime:JavaType
uid: "com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus"
fullName: "com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus"
name: "PrivateEndpointConnectionStatus"
nameWithType: "PrivateEndpointConnectionStatus"
summary: "Gets or sets the status."
inheritances:
- "<xref href=\"java.lang.Object?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"com.azure.core.util.ExpandableStringEnum?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
inheritedClassMethods:
- classRef: "<xref href=\"com.azure.core.util.ExpandableStringEnum?alt=com.azure.core.util.ExpandableStringEnum&text=ExpandableStringEnum\" data-throw-if-not-resolved=\"False\" />"
  methodsRef:
  - "<xref href=\"com.azure.core.util.ExpandableStringEnum.<T>fromString(java.lang.String,java.lang.Class<T>)?alt=com.azure.core.util.ExpandableStringEnum.<T>fromString&text=<T>fromString\" data-throw-if-not-resolved=\"False\" />"
  - "<xref href=\"com.azure.core.util.ExpandableStringEnum.<T>values(java.lang.Class<T>)?alt=com.azure.core.util.ExpandableStringEnum.<T>values&text=<T>values\" data-throw-if-not-resolved=\"False\" />"
  - "<xref href=\"com.azure.core.util.ExpandableStringEnum.equals(java.lang.Object)?alt=com.azure.core.util.ExpandableStringEnum.equals&text=equals\" data-throw-if-not-resolved=\"False\" />"
  - "<xref href=\"com.azure.core.util.ExpandableStringEnum.hashCode()?alt=com.azure.core.util.ExpandableStringEnum.hashCode&text=hashCode\" data-throw-if-not-resolved=\"False\" />"
  - "<xref href=\"com.azure.core.util.ExpandableStringEnum.toString()?alt=com.azure.core.util.ExpandableStringEnum.toString&text=toString\" data-throw-if-not-resolved=\"False\" />"
- classRef: "java.lang.<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html\">Object</a>"
  methodsRef:
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#clone--\">clone</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#finalize--\">finalize</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#getClass--\">getClass</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#notify--\">notify</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#notifyAll--\">notifyAll</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#wait--\">wait</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#wait-long-\">wait</a>"
  - "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#wait-long-int-\">wait</a>"
syntax: "public final class **PrivateEndpointConnectionStatus**</br> extends <xref href=\"com.azure.core.util.ExpandableStringEnum?alt=com.azure.core.util.ExpandableStringEnum&text=ExpandableStringEnum\" data-throw-if-not-resolved=\"False\" />&lt;<xref href=\"com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus?alt=com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus&text=PrivateEndpointConnectionStatus\" data-throw-if-not-resolved=\"False\" />&gt;"
constructors:
- uid: "com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus.PrivateEndpointConnectionStatus()"
  fullName: "com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus.PrivateEndpointConnectionStatus()"
  name: "PrivateEndpointConnectionStatus()"
  nameWithType: "PrivateEndpointConnectionStatus.PrivateEndpointConnectionStatus()"
  syntax: "public PrivateEndpointConnectionStatus()"
fields:
- uid: "com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus.APPROVED"
  fullName: "com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus.APPROVED"
  name: "APPROVED"
  nameWithType: "PrivateEndpointConnectionStatus.APPROVED"
  summary: "Static value Approved for Private<wbr>Endpoint<wbr>Connection<wbr>Status."
  modifiers:
  - "static"
  - "final"
  field:
    type: "<xref href=\"com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus?alt=com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus&text=PrivateEndpointConnectionStatus\" data-throw-if-not-resolved=\"False\" />"
  syntax: "public static final PrivateEndpointConnectionStatus APPROVED"
  desc: "Static value Approved for PrivateEndpointConnectionStatus."
- uid: "com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus.DISCONNECTED"
  fullName: "com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus.DISCONNECTED"
  name: "DISCONNECTED"
  nameWithType: "PrivateEndpointConnectionStatus.DISCONNECTED"
  summary: "Static value Disconnected for Private<wbr>Endpoint<wbr>Connection<wbr>Status."
  modifiers:
  - "static"
  - "final"
  field:
    type: "<xref href=\"com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus?alt=com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus&text=PrivateEndpointConnectionStatus\" data-throw-if-not-resolved=\"False\" />"
  syntax: "public static final PrivateEndpointConnectionStatus DISCONNECTED"
  desc: "Static value Disconnected for PrivateEndpointConnectionStatus."
- uid: "com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus.PENDING"
  fullName: "com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus.PENDING"
  name: "PENDING"
  nameWithType: "PrivateEndpointConnectionStatus.PENDING"
  summary: "Static value Pending for Private<wbr>Endpoint<wbr>Connection<wbr>Status."
  modifiers:
  - "static"
  - "final"
  field:
    type: "<xref href=\"com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus?alt=com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus&text=PrivateEndpointConnectionStatus\" data-throw-if-not-resolved=\"False\" />"
  syntax: "public static final PrivateEndpointConnectionStatus PENDING"
  desc: "Static value Pending for PrivateEndpointConnectionStatus."
- uid: "com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus.REJECTED"
  fullName: "com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus.REJECTED"
  name: "REJECTED"
  nameWithType: "PrivateEndpointConnectionStatus.REJECTED"
  summary: "Static value Rejected for Private<wbr>Endpoint<wbr>Connection<wbr>Status."
  modifiers:
  - "static"
  - "final"
  field:
    type: "<xref href=\"com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus?alt=com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus&text=PrivateEndpointConnectionStatus\" data-throw-if-not-resolved=\"False\" />"
  syntax: "public static final PrivateEndpointConnectionStatus REJECTED"
  desc: "Static value Rejected for PrivateEndpointConnectionStatus."
methods:
- uid: "com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus.fromString(java.lang.String)"
  fullName: "com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus.fromString(String name)"
  name: "fromString(String name)"
  nameWithType: "PrivateEndpointConnectionStatus.fromString(String name)"
  summary: "Creates or finds a Private<wbr>Endpoint<wbr>Connection<wbr>Status from its string representation."
  modifiers:
  - "static"
  parameters:
  - description: "a name to look for."
    name: "name"
    type: "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/lang/String.html\">String</a>"
  syntax: "public static PrivateEndpointConnectionStatus fromString(String name)"
  desc: "Creates or finds a PrivateEndpointConnectionStatus from its string representation."
  returns:
    description: "the corresponding PrivateEndpointConnectionStatus."
    type: "<xref href=\"com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus?alt=com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus&text=PrivateEndpointConnectionStatus\" data-throw-if-not-resolved=\"False\" />"
- uid: "com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus.values()"
  fullName: "com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus.values()"
  name: "values()"
  nameWithType: "PrivateEndpointConnectionStatus.values()"
  summary: "Gets known Private<wbr>Endpoint<wbr>Connection<wbr>Status values."
  modifiers:
  - "static"
  syntax: "public static Collection<PrivateEndpointConnectionStatus> values()"
  desc: "Gets known PrivateEndpointConnectionStatus values."
  returns:
    description: "known PrivateEndpointConnectionStatus values."
    type: "<a href=\"https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html\">Collection</a>&lt;<xref href=\"com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus?alt=com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus&text=PrivateEndpointConnectionStatus\" data-throw-if-not-resolved=\"False\" />&gt;"
type: "class"
desc: "Gets or sets the status."
metadata: {}
package: "com.azure.resourcemanager.recoveryservicesbackup.models"
artifact: com.azure.resourcemanager:azure-resourcemanager-recoveryservicesbackup:1.0.0-beta.6
