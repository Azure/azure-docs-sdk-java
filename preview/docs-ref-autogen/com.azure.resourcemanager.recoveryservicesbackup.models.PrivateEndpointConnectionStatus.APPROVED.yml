### YamlMime:JavaMember
uid: "com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus.APPROVED*"
fullName: "com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus.APPROVED"
name: "APPROVED"
nameWithType: "PrivateEndpointConnectionStatus.APPROVED"
members:
- uid: "com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus.APPROVED"
  fullName: "com.azure.resourcemanager.recoveryservicesbackup.models.PrivateEndpointConnectionStatus.APPROVED"
  name: "APPROVED"
  nameWithType: "PrivateEndpointConnectionStatus.APPROVED"
  summary: "Static value Approved for PrivateEndpointConnectionStatus."
  syntax: "public static final PrivateEndpointConnectionStatus APPROVED"
type: "field"
metadata: {}
package: "com.azure.resourcemanager.recoveryservicesbackup.models"
artifact: com.azure.resourcemanager:azure-resourcemanager-recoveryservicesbackup:1.0.0-beta.2
